pkg/lib/packagekit.js

```diff
@@ -479,3 +522,196 @@ export function getBackendName() {
+}
+
+/**
+ * Remove packages by their names.
+ *
+ * @param {?string[] | undefined} pkgnames - packages to remove
+ * @param {?() => void} progress_cb - progress callback
+ */
+export async function remove_packages(pkgnames, progress_cb) {
+    const ids = [];
+
+    await cancellableTransaction("Resolve", [Enum.FILTER_NOT_SOURCE | Enum.FILTER_INSTALLED | Enum.FILTER_NOT_SOURCE, pkgnames], progress_cb,
+                                 {
+                                     Package: (_info, package_id) => ids.push(package_id),
+                                 });
+
+    if (ids.length === 0)
+        return Promise.resolve();
+
+    return cancellableTransaction("RemovePackages", [0, ids, true, false], progress_cb);
+}
+
+/**
+ * @param {string[]} pkgnames - packages to install
+ * @param {?() => void} progress_cb - optional progress callback
+ */
+export async function install_packages(pkgnames, progress_cb) {
+    const flags = Enum.FILTER_ARCH | Enum.FILTER_NOT_SOURCE | Enum.FILTER_NEWEST;
+    const ids = [];
+
+    await cancellableTransaction("Resolve", [flags | Enum.FILTER_NOT_INSTALLED, Array.from(pkgnames)], progress_cb,
+                                 {
+                                     Package: (_info, package_id) => ids.push(package_id),
+                                 });
+
+    if (ids.length === 0)
+        return Promise.reject(new TransactionError("not-found", "Can't resolve package(s)"));
+    else
+        return cancellableTransaction("InstallPackages", [0, ids], progress_cb)
+                .catch(ex => {
+                    if (ex.code != Enum.ERROR_ALREADY_INSTALLED)
+                        return Promise.reject(ex);
+                });
+}
+
+/**
+ * On Debian the update_text starts with "== version ==" which is
+ * redundant; we don't want Markdown headings in the table
+ *
+ * @param {string} text - update_text to filter
+ */
+function removeHeading(text) {
+    if (text)
+        return text.trim().replace(/^== .* ==\n/, "")
+                .trim();
+    return text;
+}
+
+// parse CVEs from an arbitrary text (changelog) and return URL array
+function parseCVEs(text) {
+    if (!text)
+        return [];
+
+    const cves = text.match(/CVE-\d{4}-\d+/g);
+    if (!cves)
+        return [];
+    return cves.map(n => "https://www.cve.org/CVERecord?id=" + n);
+}
+
+function deduplicate(list) {
+    return [...new Set(list)].sort();
+}
+
+/** @returns {Promise<void>} */
+function loadUpdateDetailsBatch(pkg_ids, update_details, progress_cb) {
+    return cancellableTransaction("GetUpdateDetail", [pkg_ids], progress_cb, {
+        UpdateDetail: (packageId, _updates, _obsoletes, vendor_urls, bug_urls, cve_urls, _restart,
+            update_text, changelog /* state, issued, updated */) => {
+            const u = update_details[packageId];
+            if (!u) {
+                console.warn("Mismatching update:", packageId);
+                return;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22498-f9022dfb-20251118-164913-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)